app/uk/gov/hmrc/economiccrimelevyregistration/controllers/RegistrationSubmissionController.scala

```diff
@@ -63,13 +63,14 @@ class RegistrationSubmissionController @Inject() (
             additionalInfo =>
               registrationValidationService.validateRegistration(registration, additionalInfo) match {
                 case Valid(Left(eclSubscription)) =>
-                  subscriptionService.subscribeToEcl(eclSubscription, registration).map { response =>
-                    nrsService.submitToNrs(
-                      registration.base64EncodedFields.flatMap(_.nrsSubmissionHtml),
-                      response.success.eclReference,
-                      appConfig.eclFirstTimeRegistrationNotableEvent
-                    )
-                    Ok(Json.toJson(response.success))
+                  subscriptionService.subscribeToEcl(eclSubscription, registration, additionalInfo.liabilityYear).map {
```

Comment:
Why do you pass thge luability year when it is part of the `eclSubscription`?

Reply:
We have the liabilityStartDate in eclSubscription but since we just need the year which we have stored in AdditionalInfo so I just passed that instead of deriving it from the string